### ISSUE TITLE
Add ability to export textures included in .material files as PNG/JPEG

### ISSUE THREAD

**Issue by JohnGabrielUK** [godotengine/godot-proposals#3443]
### Describe the project you are working on

A 3D game using high-resolution textures.

### Describe the problem or limitation you are having in your project

In my game, some of my materials have their albedo/normal/etc textures baked into the `.material` file. It would be nice to be able to retrieve those textures in PNG or JPEG form from the material without having to track down the original image - as it is, I can only export them in TRES or RES format. The reason I would like to do this is because, with the image included, the material file is several megabytes big, which is much larger than the textures would be in PNG/JPEG format.

![image](https://user-images.githubusercontent.com/5272862/137914028-7402263a-15b3-4c39-bca1-be190ab30d54.png)
![image](https://user-images.githubusercontent.com/5272862/137914070-35420f5c-04e9-4310-861b-1507a534d50b.png)
![image](https://user-images.githubusercontent.com/5272862/137914111-289e94d1-80b5-43f0-91a0-be6facec73c1.png)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

The feature adds options for PNG and JPEG to the existing TRES and RES formats shown in the screenshot above.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

From the UI perspective, it would just be a case of adding PNG/JPEG options to the format box in the "Save Resource As" dialog.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

You could write a Godot script that could read the ImageTexture in those materials and save it as a PNG, but this is not very convenient.

### Is there a reason why this should be core and not an add-on in the asset library?

I think that this should be in core because it fixes a problem of textures being "locked" into material files, and thus only useable in Godot, and that the fix would be very lightweight (since Godot already has the ability to write PNG/JPEG files, so it would just be a case of applying that functionality in this situation).

**Comment by fire** [comment]
Exporting as Openexr has some use cases too.

**Comment by Calinou** [comment]
> Exporting as Openexr has some use cases too.

Indeed, HDR textures should be exported as `.exr`. This is feasible in editor builds with `Image.save_exr()`.
